Remove PACKED from ht_tag_flags_s struct.

Some architectures like Apple Silicon do not allow unaligned pointers; this
fixes compilation on them. I could also not find a reason why this struct would
need to be packed, and the resulting code seems to work fine without.

Fixes:

ld: pointer not aligned in '_coff_characteristics'+0x1

https://github.com/sebastianbiallas/ht/pull/31
--- httag.h.orig
+++ httag.h
@@ -71,7 +71,7 @@ struct ht_tag_flags {
 struct ht_tag_flags_s {
 	char bitidx;
 	const char *desc;
-} PACKED;
+};
 
 /* GROUP-TAG */
 #define HT_TAG_GROUP				0x03
